Signal/src/ViewControllers/SignalAttachment.swift

```diff
@@ -204,6 +216,12 @@ class SignalAttachment: NSObject {
     // Returns the file extension for this attachment or nil if no file extension
     // can be identified.
     var fileExtension: String? {
+        if let filename = filename {
+            let fileExtension = (filename as NSString).pathExtension
+            if fileExtension.characters.count > 0 {
+                return fileExtension
+            }
+        }
```

Comment:
A small but important fix - prefer to determine the file extension from the filename than the mimetype.  This will avoid some cases where attachments will be relabeled.  We should trust the file extension we have.